src/hosts.cpp

```diff
@@ -85,6 +88,43 @@ code hosts::fetch(address& out) const
     ///////////////////////////////////////////////////////////////////////////
 }
 
+code hosts::fetch(address::list& out) const
+{
+    if (disabled_)
+        return error::not_found;
+
+    size_t out_count = 0;
+
+    // Critical Section
+    ///////////////////////////////////////////////////////////////////////////
+    {
+        shared_lock lock(mutex_);
```

Comment:
mutex_.lock_shared();

Reply:
Is it style that prefers the explicit lock/unlock instead of keeping the scoped locks?

Reply:
Yes, it reduces nesting, as well as line length growth (which we try to limit), saves a line of code, and in the majority of cases there is a need to change lock state using an upgradeable. There are no unqualified scopes anywhere in the codebase.

Reply:
Ok.  In this case, it will increase lines of code though, no?

```
    mutex_.lock_shared();

    if (stopped_)
    //{
    //  mutex_.unlock_shared();
        return error::service_stopped;
    //}

    if (buffer_.empty())
    //{
    //  mutex_.unlock_shared();
        return error::not_found;
    //}

... etc ...
```

Reply:
Yes, of course, missed that.

Reply:
Ok